lib/spack/spack/spec_parser.py

```diff
@@ -528,6 +563,11 @@ def parse(self):
         if "deptypes" in attributes:
             deptype_string = attributes.pop("deptypes")
             attributes["depflag"] = spack.deptypes.canonicalize(deptype_string)
+
+        # Turn "when" into a spec
+        if "when" in attributes:
+            attributes["when"] = spack.spec.Spec(attributes["when"][0])
```

Comment:
Isn't the Spec constructor just calling some function in the parser? Can we use `parse_one_or_raise`?